From a052913dd020b73a0881e9f5697e03885b3ca298 Mon Sep 17 00:00:00 2001
From: Christian Brauner <christian.brauner@canonical.com>
Date: Sat, 19 Nov 2016 05:45:01 +0100
Subject: [PATCH] utils: bugfixes

- simply check /proc/self/ns
- use #define to prevent gcc & clang to use a VLA

Signed-off-by: Christian Brauner <christian.brauner@canonical.com>
---
 src/lxc/utils.c  |    9 +++++----
 2 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/src/lxc/utils.c b/src/lxc/utils.c
index 70f5aeb..9198340 100644
--- a/src/lxc/utils.c
+++ b/src/lxc/utils.c
@@ -1972,17 +1972,18 @@ int lxc_append_string(char ***list, char *entry)
 int lxc_preserve_ns(const int pid, const char *ns)
 {
 	int ret;
-	size_t len = 5 /* /proc */ + 21 /* /int_as_str */ + 3 /* /ns */ + 20 /* /NS_NAME */ + 1 /* \0 */;
-	char path[len];
+/* 5 /proc + 21 /int_as_str + 3 /ns + 20 /NS_NAME + 1 \0 */
+#define __NS_PATH_LEN 50
+	char path[__NS_PATH_LEN];
 
 	/* This way we can use this function to also check whether namespaces
 	 * are supported by the kernel by passing in the NULL or the empty
 	 * string.
 	 */
-	ret = snprintf(path, len, "/proc/%d/ns%s%s", pid,
+	ret = snprintf(path, __NS_PATH_LEN, "/proc/%d/ns%s%s", pid,
 		       !ns || strcmp(ns, "") == 0 ? "" : "/",
 		       !ns || strcmp(ns, "") == 0 ? "" : ns);
-	if (ret < 0 || (size_t)ret >= len)
+	if (ret < 0 || (size_t)ret >= __NS_PATH_LEN)
 		return -1;
 
 	return open(path, O_RDONLY | O_CLOEXEC);
-- 
1.7.9.5

